qwerty_mlir/lib/Qwerty/IR/QwertyOps.cpp

```diff
@@ -843,11 +843,31 @@ mlir::Type FuncOp::cloneTypeWith(mlir::TypeRange inputs,
 mlir::LogicalResult FuncOp::verifyBody() {
     size_t n_func_args = getQwertyFuncType().getFunctionType().getInputs().size();
     // Extra arguments are captures
-    if (getBody().getNumArguments() >= n_func_args) {
-        return mlir::success();
-    } else {
+    if (getBody().getNumArguments() < n_func_args) {
         return mlir::failure();
     }
+
+    // Now we just check if all values that need to be linear are, in fact, linear.
+    for (mlir::Block& b : getBody()) {
```

Comment:
```suggestion
    for (mlir::Block &b : getBody()) {
```

Reply:
My reasoning is that `x` has type `int &`, and `y` has type `int` in the following code:

```cpp
int& x, y;
```